cogs/roles.py

```diff
@@ -1,94 +1,30 @@
-    async def buttonroles(self, ctx: commands.Context):
-        """
-        Send a message with buttons that add roles.
-        """ 
-        raise NotImplementedError('Command requires implementation and permission set-up.')
-
-    @commands.hybrid_group()
-    async def cassowary(self, ctx: commands.Context):
-        """
-        Commands for handling cassowaries.
-        """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
-
-    @cassowary.command()
-    async def create(self, ctx: commands.Context):
-        """
-        Create a cassowary.
-        """ 
-        raise NotImplementedError('Command requires implementation and permission set-up.')
-
-    @cassowary.command()
-    async def delete(self, ctx: commands.Context):
-        """
-        Delete a cassowary.
-        """ 
-        raise NotImplementedError('Command requires implementation and permission set-up.')
-
-    @cassowary.command()
-    async def list(self, ctx: commands.Context):
-        """
-        List all cassowaries.
-        """ 
-        raise NotImplementedError('Command requires implementation and permission set-up.')
-
+    
+    cog_check = cogify(admin_command())
 
     @commands.hybrid_command()
-    async def giverole(self, ctx: commands.Context):
+    async def giverole(self, ctx: commands.Context, existing_role: discord.Role, new_role: discord.Role):
         """
         Assign a new role to all members with a specific role.
-        """ 
-        raise NotImplementedError('Command requires implementation and permission set-up.')
-
-
-    @commands.hybrid_group()
-    async def reaction(self, ctx: commands.Context):
         """
-        Commands for handling reaction roles.
-        """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
-
-    
-    @reaction.command(aliases=['add'])
-    async def create(self, ctx: commands.Context):
-        """
-        Create a reaction role.
-        """ 
-        raise NotImplementedError('Command requires implementation and permission set-up.')
-    
-    @reaction.command(aliases=['remove'])
-    async def delete(self, ctx: commands.Context):
-        """
-        Delete a reaction role.
-        """ 
-        raise NotImplementedError('Command requires implementation and permission set-up.')
-
+        if existing_role is None or new_role is None:
+            return await ctx.send("Please provide both the existing and new role.")
+        role_members = [member for member in ctx.guild.members if existing_role in member.roles]
+        for member in role_members:
+            try:
+                await member.add_roles(new_role)
+            except discord.Forbidden:
+                return await ctx.send("I do not have permission to add roles to members.")
+            except discord.HTTPException:
+                await ctx.send(f"Could not add role to {member.name}")
+        await ctx.send("Successfully added roles.")
```

Comment:
Could be useful to add the number of users that the role was added to in the success message.

Reply:
A log info message would be good (same message as this with number of users and the role IDs)